packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -254,7 +254,10 @@ class TextSelectionOverlay {
        assert(context != null),
        _value = value {
     final OverlayState overlay = Overlay.of(context);
-    assert(overlay != null);
+    assert(overlay != null,
+      'No Overlay widget exists above $context.\n'
+      'Usually the Navigator created by WidgetsApp provides the overlay. Perhaps your '
+      'app content was created above the Navigator with the WidgetsApp builder parameter.');
```

Comment:
This looks good, is there anything we can suggest for a fix here?  Nothing good is coming to mind for me though.

Reply:
Developers seem to occasionally and incorrectly use the WidgetsApp And MaterialApp `builder` parameter. It's generic name and the fact that it appears first in the docs probably don't help matters. I was hoping that the error message would encourage developers to look for this failure mode. The "fix" for the problem depends on if they're using the builder parameter intentionally or not.